File: app/src/main/java/com/microlabs/trallet/adapter/CategorySpinnerAdapter.java
package com.microlabs.trallet.adapter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.microlabs.trallet.R;
import com.microlabs.trallet.base.RealmHelper;
import com.microlabs.trallet.model.Category;

import java.util.ArrayList;

/**
 * Created by Nicholas on 3/13/2017.
 */

public class CategorySpinnerAdapter extends ArrayAdapter<Category>{

    private ArrayList<Category> mCategory;
    private LayoutInflater inflater;

    public CategorySpinnerAdapter(Context context, int resource, ArrayList<Category> categories) {
        super(context, resource, categories);
        mCategory = categories;
        inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }

    @Override
    public View getDropDownView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        View v = inflater.inflate(R.layout.item_category,parent,false);
        TextView lblCategory = (TextView) v.findViewById(R.id.lblCategoryText);
        ImageView imgCategory = (ImageView) v.findViewById(R.id.imgCategory);
        lblCategory.setText(mCategory.get(position).getName());
        imgCategory.setImageResource(setImage(position));
        return v;
    }

    private int setImage(int position) {
        switch (position) {
            case 0:
                return R.drawable.ic_local_dining_black_24dp;
            case 1:
                return R.drawable.ic_local_taxi_black_24dp;
            case 2:
                return R.drawable.ic_shopping_basket_black_24dp;
            default:
                return R.drawable.ic_local_atm_black_24dp;
        }
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        View v = inflater.inflate(R.layout.item_category,parent,false);
        TextView lblCategory = (TextView) v.findViewById(R.id.lblCategoryText);
        ImageView imgCategory = (ImageView) v.findViewById(R.id.imgCategory);
        lblCategory.setText(mCategory.get(position).getName());
        imgCategory.setImageResource(setImage(position));
        return v;
    }

    public int getPosition(int catId) {
        return super.getPosition(RealmHelper.getCategoryById(catId));
    }
}

File: app/src/main/java/com/microlabs/trallet/adapter/CategorySpinnerAdapter.kt
package com.microlabs.trallet.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import com.microlabs.trallet.R
import com.microlabs.trallet.model.Category
import kotlinx.android.synthetic.main.item_category.view.*

/**
 * Created by Nicholas on 3/13/2017.
 *
 * RCategory Spinner Custom Adapter
 */

class CategorySpinnerAdapter(context: Context, resource: Int, private val categoryList: List<Category>) : ArrayAdapter<Category>(context, resource, categoryList) {

    private val inflater: LayoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    private var itemView: View? = null

    override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
        val v = inflater.inflate(R.layout.item_category, parent, false)
        v.lblCategoryText.text = categoryList[position].name
        v.imgCategory.setImageResource(categoryList[position].imgId)
        return v
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        if (itemView == null) {
            itemView = inflater.inflate(R.layout.item_category, parent, false)
        }
        itemView!!.lblCategoryText.text = categoryList[position].name
        itemView!!.imgCategory.setImageResource(categoryList[position].imgId)
        return itemView!!
    }

    fun getPositionById(categoryId: Int): Int {
        return categoryList.indexOfFirst { it.id == categoryId }
    }
}
